### DISCUSSION TITLE
v1.7.0-beta

### DISCUSSION THREAD

**Discussion by JAGFx** [JAGFx/ets2-dashboard-skin#98] [General]
# Note

**THIS IS A PRE-RELEASE**: Be aware that this is a beta. You can get some errors or crashes. If you find one, write on the discussion open for it. ;). You can also keep an eye on the #97 issue.

## Documentation

Firstly, I made some changes in the documentation. Obviously to adapt to the new functionality, but also to have something cleaner and simpler. 

In this case, the known issues will be reworked, and the bug report template will focus on the essential things.
The custom map section will also be simpler with precise step.

## The main purpose

In this version, the server has been reworked. This was necessary to get the configuration for a specific game.

Now you have three files:
- One for the minimal configuration
- One for Eurotrucksimator 2 configuration
- One for Americiantruck simulator

This redesign was an opportunity to introduce some logs to help you and me identify problems ;) You can read the section [Identify a problem](https://github.com/JAGFx/ets2-dashboard-skin/blob/master/CONTRIBUTING.md#identify-a-problem)

# Breaking changes

- The configuration system is redesigned. The old configuration file is no longer used. Please refer to the change log for more information.
- The host `ets2.jagfx.fr` is deprecated and will be removed in the future.

To see all others changes, see this [readme](https://github.com/JAGFx/ets2-dashboard-skin/blob/master/doc/BREAKING_CHANGES.md)

# Features

- Load a specific configuration for ETS2 or ATS
- Add logs for server in `logs` folder
- Reword some documentation
- Reword bug report template

# Changeslogs

0cc37db - :bookmark: release v1.7.0-beta (JAGFx)
b7102dd - :rotating_light: #83 correct linter warning (JAGFx)
7a31b3e - :bug: #83 adjust authorized host (JAGFx)
e8bc83f - :memo: update tags on main readme (JAGFx)
eb2b7f2 - :memo: update some documentation (JAGFx)
b734a48 - :memo: update bug report template (SMITH Emmanuel)
6c27fed - :memo: correct github bug template (JAGFx)
cda971c - :memo: update contribution documentation (JAGFx)
1fa804c - :memo: update documentation (JAGFx)
f7268f9 - :memo: update bug declaration (JAGFx)
a642944 - :wastebasket: [feat/config] remove dead code (JAGFx)
e378a4c - :rotating_light: [feat/config] #83 apply linter fix (JAGFx)
43b601f - :bug: [feat/config] #83 fix saving config and production server (JAGFx)
dc552df - :wastebasket: [feat/config] #77 minor typo correction + add translation + breaking changes (JAGFx)
77c9821 - :children_crossing: [feat/config] adjust the way to use custom remote tile host (JAGFx)
597f116 - :truck: [feat/config] #83 move scs sdk files in lib directory (JAGFx)
61f7915 - :bug: [feat/config] #77 enable configuration upload validation (JAGFx)
774da12 - :bug: [feat/config] #83 update error handler (JAGFx)
9646176 - :bug: [feat/config] #77 correct download config (JAGFx)
e290415 - :bug: [feat/config] #83 remove untracked file (JAGFx)
a6f05fc - :sparkles: [feat/config] #83 add logger (JAGFx)
b6884bc - :construction_worker: [feat/config] #83 add ignore file (JAGFx)
377176a - :memo: [feat/config] #77 update documentation (JAGFx)
08c1938 - :memo: [feat/config] #77 update documentation (JAGFx)
630de4b - :wastebasket: [feat/config] #83 remove unused folder (JAGFx)
01a4f39 - :construction_worker: [feat/config] #83 correct builder (JAGFx)
5ec9e77 - :rotating_light: [feat/config] #83 apply linter recommendations (JAGFx)
7dfa23b - :recycle: [feat/config] #83 factorization of default config (JAGFx)
faa9fbd - :sparkles: [feat/config] #83 add promise for server (JAGFx)
fc02a57 - Merge branch 'master' into feat/config (JAGFx)
530bce6 - :bug: [feat/config] #83 correct lib to give config game (JAGFx)
56708aa - :pencil: update the minimal requirements (SMITH Emmanuel)
f25aee1 - :twisted_rightwards_arrows: Add support for iOS full screen web clip, clean up grammar (#90) (TheGamer1002)
c86a56c - :pencil: update known issues (SMITH Emmanuel)
a5df28a - :sparkles: [feat/config] #77 add true loading of convocation for dev server (JAGFx)
ce5ec43 - :sparkles: [feat/config] #77 add load ats config on dev environment (JAGFx)
3f4da8c - :construction: [feat/config] #77 correct get config from server (SMITH Emmanuel)
ef51b80 - :construction: [feat/config] #77 rework server (JAGFx)
e06a133 - :sparkles: [feat/config] #77 update configuration files (SMITH Emmanuel)
d1c277b - Merge branch 'master' into feat/config (JAGFx)
e8ae1c3 - :memo: add credits (SMITH Emmanuel)
5894084 - :sparkles: [feat/config] #77 add download and upload of new configuration file (SMITH Emmanuel)
e3e6c48 - Merge branch 'master' into feat/config (JAGFx)
aedbb00 - :memo: update readme (SMITH Emmanuel)
71afe69 - Fixed small typo, some changes and added missing line (#88) (Jonny_Bro)
a53f05a - :construction: #77 [feat/config] rework the loading config (SMITH Emmanuel)
ccfc79e - Merge branch 'master' into feat/config (JAGFx)
969225c - :construction: #77 [feat/config] auto change config for a game (SMITH Emmanuel)

<hr /><em>This discussion was created from the release <a href='https://github.com/JAGFx/ets2-dashboard-skin/releases/tag/v1.7.0-beta'>v1.7.0-beta</a>.</em>

**Comment by bmwGTR** [comment]
Hello,

here are the bugs I found:
1.) I can't switch the "Damage accurate" from Damage diagnostic to Route adivsor on Config screen
2.) If I download Game config, it is named "config.undefined.json" always (no matter if ets2 or ats is running)
3.) the config is only switched when dashboard is restarted. if I shutdown ETS2 and start ATS (without dashboard restart) it still shows the old config

Server log file:
[server.log](https://github.com/JAGFx/ets2-dashboard-skin/files/7883886/server.log)

**Comment by JAGFx** [comment]
@bmwGTR Hi,

Just a note for the third point. Currently, I'm not reloading the configuration when the game is switched.
Nevertheless, it's a great feature that can be part of a specific feature request.
